pkg/daemon/server/service/pipeline_metrics_query.go

```diff
@@ -144,17 +144,30 @@ func (ps *PipelineMetadataQuery) GetVertexMetrics(ctx context.Context, req *daem
 	if abstractVertex.IsASource() {
 		bufferList = append(bufferList, req.GetVertex())
 	}
-	partitionPendingInfo := ps.getPending(ctx, req)
```

Comment:
The implementation of this function should not be changed. The pending information returned in this implementation is a set of average values in the past period of times, and we use one of the average values to compute desired replicas. 

To find the instant buffer pending number, we can use GetPipelineBuffer().

Reply:
Reverted deamon code 👍 

Using GetPipelineBuffer() to fetch relevant buffers of the vertex now